zkvm/src/signature/multikey.rs

```diff
@@ -0,0 +1,58 @@
+use crate::errors::VMError;
+use crate::signature::VerificationKey;
+use crate::transcript::TranscriptProtocol;
+use curve25519_dalek::ristretto::RistrettoPoint;
+use curve25519_dalek::scalar::Scalar;
+use merlin::Transcript;
+
+#[derive(Clone)]
+pub struct Multikey {
+    transcript: Transcript,
+    X_agg: VerificationKey,
+}
+
+impl Multikey {
+    pub fn new(pubkeys: Vec<VerificationKey>) -> Result<Self, VMError> {
+        // Create transcript for Multikey
+        let mut transcript = Transcript::new(b"ZkVM.aggregated-key");
+
+        // Hash in pubkeys
+        // L = H(X_1 || X_2 || ... || X_n)
+        for X_i in &pubkeys {
+            transcript.commit_point(b"X_i.L", &X_i.0);
```

Comment:
is there a better label than jargon from paper? E.g. `"P"` for "pubkey"? Even `"X_i"` is better than `"X_i.L"` since L is the transcript state itself.

Reply:
ok. I originally called it `X_i.L` to differentiate it from the `X_i` that goes into the transcript for `factor_for_key`. I can change it to `P`!